lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
 
-  defstruct port: nil,
-            requests: []
+  @typedoc "Setup options."
+  @type options :: map 
 
+  @typedoc "Interface name"
+  @type ifname :: binary
```

Comment:
:+1: I use `binary` just personally out of habit since it is locally typechecked. Remote types have problems in my editor, so i stick to `binary` if i can. No problems with changing it.